homeassistant/components/eheimdigital/config_flow.py

```diff
@@ -0,0 +1,93 @@
+"""Config flow for EHEIM Digital."""
+
+from __future__ import annotations
+
+from typing import Any
+
+from aiohttp import ClientError
+from eheimdigital.hub import EheimDigitalHub
+import voluptuous as vol
+
+from homeassistant.components.zeroconf import ZeroconfServiceInfo
+from homeassistant.config_entries import SOURCE_USER, ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_HOST
+from homeassistant.helpers import selector
+from homeassistant.helpers.aiohttp_client import async_create_clientsession
+
+from .const import DOMAIN
+
+CONFIG_SCHEMA = vol.Schema(
+    {vol.Required(CONF_HOST, default="eheimdigital"): selector.TextSelector()}
+)
+
+
+class EheimDigitalConfigFlowHandler(ConfigFlow, domain=DOMAIN):
+    """The EHEIM Digital config flow."""
+
+    data: dict[str, Any] = {}
+
+    async def async_step_zeroconf(
+        self, discovery_info: ZeroconfServiceInfo
+    ) -> ConfigFlowResult:
+        """Handle zeroconf discovery."""
+        self.data[CONF_HOST] = host = discovery_info.host
+
+        self._async_abort_entries_match(self.data)
+
+        hub = EheimDigitalHub(
+            session=async_create_clientsession(self.hass, base_url=f"http://{host}"),
+            loop=self.hass.loop,
+        )
+        await hub.connect()
+        await hub.update()
+        await hub.close()
```

Comment:
Why don't we have a try catch here

Reply:
I used `airgradient` as an orientation, and there it isn't done as well in the zeroconf step.

Reply:
Oh oops, that's something I should fix